File: src/nl/rubensten/texifyidea/file/ClassFileType.java
package nl.rubensten.texifyidea.file;

import com.intellij.openapi.fileTypes.LanguageFileType;
import nl.rubensten.texifyidea.LatexLanguage;
import nl.rubensten.texifyidea.TexifyIcons;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * @author Ruben Schellekens
 */
public class ClassFileType extends LanguageFileType {

    public static final ClassFileType INSTANCE = new ClassFileType();

    private ClassFileType() {
        super(LatexLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "LaTeX class file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "LaTeX class file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "cls";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return TexifyIcons.CLASS_FILE;
    }
}

File: src/nl/rubensten/texifyidea/file/ClassFileType.kt
package nl.rubensten.texifyidea.file

import com.intellij.openapi.fileTypes.LanguageFileType
import nl.rubensten.texifyidea.LatexLanguage
import nl.rubensten.texifyidea.TexifyIcons

/**
 * @author Ruben Schellekens
 */
object ClassFileType : LanguageFileType(LatexLanguage.INSTANCE) {

    override fun getName() = "LaTeX class file"

    override fun getDescription() = "LaTeX class file"

    override fun getDefaultExtension() = "cls"

    override fun getIcon() = TexifyIcons.CLASS_FILE!!
}
